vendor/engines/secure_rooms/app/services/secure_rooms/access_rules/base_rule.rb

```diff
@@ -23,19 +23,19 @@ def evaluate
       end
 
       def pass
-        Verdict.new(:pass)
+        Verdict.new(:pass, :passed, @user, @card_reader)
       end
 
-      def grant!
-        Verdict.new(:grant)
+      def grant!(reason, options = {})
+        Verdict.new(:grant, reason, @user, @card_reader, options)
       end
 
-      def pending!(reason)
-        Verdict.new(:pending, reason: reason)
+      def pending!(reason, options = {})
+        Verdict.new(:pending, reason, @user, @card_reader, options)
       end
 
-      def deny!(reason)
-        Verdict.new(:deny, reason: reason)
+      def deny!(reason, options = {})
+        Verdict.new(:deny, reason, @user, @card_reader, options)
       end
```

Comment:
I kind of want to make this like a factory or something now but I feel I'm going too far